cassandra/src/main/java/org/hibernate/ogm/datastore/cassandra/CassandraDialect.java

```diff
@@ -96,8 +112,17 @@ public LockingStrategy getLockingStrategy(Lockable lockable, LockMode lockMode)
 
 	private ResultSet bindAndExecute(Object[] columnValues, RegularStatement statement) {
 
+		PreparedStatement preparedStatement;
+		try {
+			preparedStatement = preparedStatementCache.get( statement.getQueryString() );
+			session.prepare( statement );
+		}
+		catch (ExecutionException e) {
+			log.failToPrepareCQL( statement.getQueryString(), e.getCause() );
```

Comment:
I'd prefer to either log *or* re-throw. Best IMO would be to throw a `HibernateException`, wrapping the original one and containing the query, thrown via our common `Log` class pattern (so it gets a unique code, too).